Source/SwiftLintCore/Models/Linter.swift

```diff
@@ -16,47 +17,87 @@ private struct LintResult {
 }
 
 private extension Rule {
-    static func superfluousDisableCommandViolations(regions: [Region],
-                                                    superfluousDisableCommandRule: SuperfluousDisableCommandRule?,
-                                                    allViolations: [StyleViolation]) -> [StyleViolation] {
+    func superfluousDisableCommandViolations(regions: [Region],
+                                             superfluousDisableCommandRule: SuperfluousDisableCommandRule?,
+                                             allViolations: [StyleViolation]) -> [StyleViolation] {
         guard regions.isNotEmpty, let superfluousDisableCommandRule else {
             return []
         }
 
-        let regionsDisablingCurrentRule = regions.filter { region in
-            region.isRuleDisabled(self.init())
-        }
         let regionsDisablingSuperfluousDisableRule = regions.filter { region in
             region.isRuleDisabled(superfluousDisableCommandRule)
         }
 
-        return regionsDisablingCurrentRule.compactMap { region -> StyleViolation? in
-            let isSuperfluousRuleDisabled = regionsDisablingSuperfluousDisableRule.contains {
-                $0.contains(region.start)
+        let identifiersWithRegions: [(String, [Region])] = {
+            if let customRules = self as? CustomRules {
```

Comment:
`CustomRules` is a `Rule` as well. Can we just override this method for `CustomRules` instead of branching here?

Reply:
yes, that would probably be better - I'll look at re-arranging this

Reply:
So I've abtracted this logic into a `func disabledRuleIdentifiersWithRegions(regions: [Region]) -> [(String, [Region])]` method, which is implemented in an extension of `Rule`, and then also implemented on `CustomRules`.